gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java

```diff
@@ -648,6 +650,7 @@ public void testInvalidChangeRequest() {
         assertEquals(400, ex.code());
       }
     } finally {
+      assertTrue(recordAdded);
```

Comment:
I was thinking more of:
```java
if (recordAdded) {
  ChangeRequestInfo deletion = ChangeRequest.builder().delete(validA).build();
  ChangeRequest request = zone.applyChangeRequest(deletion);
  waitForChangeToComplete(zone.name(), request.id());
}
zone.delete();
```

to allow cleanup while, hopefully, propagating the original error.